core/include/traccc/seeding/seeding_algorithm.hpp

```diff
@@ -28,7 +28,10 @@ class seeding_algorithm : public algorithm<seed_collection_types::host(
     ///
     /// @param mr The memory resource to use
     ///
-    seeding_algorithm(vecmem::memory_resource& mr);
+    seeding_algorithm(const seedfinder_config& finder_config,
+                      const spacepoint_grid_config& grid_config,
+                      const seedfilter_config& filter_config,
+                      vecmem::memory_resource& mr);
```

Comment:
If you want to avoid modifying the client code in this MR, you should add this as an additional constructor on top of the existing one, which only takes a memory resource. (Default function arguments don't really play here, as I would also very much prefer the config objects coming before the memory resource when we want to define everything.)